staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -133,23 +117,55 @@ type ExposeServiceOptions struct {
 
 	Recorder genericclioptions.Recorder
 	genericiooptions.IOStreams
+	ExposeInputOptions ExposeInputMetadata
 }
 
-// NewExposeServiceOptions creates a new ExposeServiceOptions and return a pointer to the
-// struct
-func NewExposeServiceOptions(ioStreams genericiooptions.IOStreams) *ExposeServiceOptions {
-	return &ExposeServiceOptions{
-		RecordFlags: genericclioptions.NewRecordFlags(),
-		PrintFlags:  genericclioptions.NewPrintFlags("exposed").WithTypeSetter(scheme.Scheme),
+// exposeServiceFlags is a struct that contains the user input flags to the command.
+type exposeServiceFlags struct {
```

Comment:
Why is this not exported? Because there might be external consumers try to use this command, the first surface they encounter would be `ExposeServiceFlags`?

Reply:
I assume we wouldn't want external consumers to modify `exposeServiceFlags` (nor the options that are taken through it)?. In the sense if they consume as a library, they can directly set the options and if as a cli we would want them to use the cmd flags. Is that not the case?

Reply:
Usually consumers(like plugins) rely on directly to Flags(especially to use `AddFlags`)